### PR TITLE
fix(lidar_centerpoint): non-maximum suppression target decision logic

### PR DESCRIPTION
**Pull request by technolojin**
## Description

1. fix non-maximum suppression target decision logic
    Before: If both of the object class are in the target label, caluclate IoU. If not, check distance and calculate IoU if the distance is less than a threshold.
    After: If the label pair is not the same and one of them is pedestrian, do not suppress. Then, check distance and calcuclate IoU only if the distance is within a threshold.
 
2. optimize non-maximum suppression search range
   Keep the squared value of the distance threshold.

3. remove NMS type
  Since there is no other NMS option, remove the logic and parameter to select NMS type.

## Related links

NMS PR: https://github.com/autowarefoundation/autoware.universe/pull/1935

Issue: 
 - pedestrian near a car is not detected https://github.com/autowarefoundation/autoware.universe/issues/7563 

## How was this PR tested?
Performance test https://evaluation.tier4.jp/evaluation/reports/16239dd4-5213-5843-ada0-49341293bb6d?project_id=prd_jt
Execution test https://evaluation.tier4.jp/evaluation/reports/26db2be4-4fe5-5fe9-8221-320a88f318d7?project_id=prd_jt

## Notes for reviewers

Remove unused parameter https://github.com/autowarefoundation/autoware_launch/pull/1267 will be merged after this PR.

## Interface changes

Parameter `iou_nms_target_class_names` will be removed.

## Effects on system behavior

None.

### CODE REVIEW

**Comment by kminoda** [comment]
@technolojin Would you separate the PR for transfusion? I want to make the PR as small as possible (OK for image_projection_based_fusion and centerpoint as those two packages are dependent)

File: perception/autoware_image_projection_based_fusion/config/pointpainting.param.yaml [hunk 1]
```diff
@@ -9,7 +9,6 @@
     post_process_params:
       # post-process params
       circle_nms_dist_threshold: 0.3
-      iou_nms_target_class_names: ["CAR"]
```
Comment (kminoda) [inline perception/autoware_image_projection_based_fusion/config/pointpainting.param.yaml:12]: Why are we removing this parameter?

(Also, if we change this, don't we need to update "Interface changes" part?)

File: perception/autoware_lidar_centerpoint/include/autoware/lidar_centerpoint/postprocess/non_maximum_suppression.hpp [hunk 2]
```diff
@@ -28,36 +28,12 @@ namespace autoware::lidar_centerpoint
...
 using autoware_perception_msgs::msg::DetectedObject;
 
-// TODO(yukke42): now only support IoU_BEV
-enum class NMS_TYPE {
-  IoU_BEV
-  // IoU_3D
-  // Distance_2D
-  // Distance_3D
-};
-
```
Comment (kminoda) [inline perception/autoware_lidar_centerpoint/include/autoware/lidar_centerpoint/postprocess/non_maximum_suppression.hpp:38]: Why do we remove this in this PR? I kind of understand the benefit of this modification, but even so, would be better to do it in another PR.

File: perception/autoware_image_projection_based_fusion/config/pointpainting.param.yaml (hunk 1 above)
Comment (technolojin) [reply perception/autoware_image_projection_based_fusion/config/pointpainting.param.yaml:12]: > Interface changes
I mentioned this change. Thanks for remind.

The parameter seems to be implemented only for those classes, but the implementation didn't limit the NMS process
As of today, the NMS should be implemented all of the classes. Only exemption is between pedestrian and non-pedestrian.

**Comment by technolojin** [comment]
> @technolojin Would you separate the PR for transfusion? I want to make the PR as small as possible (OK for image_projection_based_fusion and centerpoint as those two packages are dependent)

@kminoda 
Then, the maintainer of the transfusion will decide if the NMS logic to be fixed or not.

File: perception/autoware_image_projection_based_fusion/config/pointpainting.param.yaml (hunk 1 above)
Comment (kminoda) [reply perception/autoware_image_projection_based_fusion/config/pointpainting.param.yaml:12]: Ah true, now make sense to me, thanks for the explanation 🙏

File: perception/autoware_lidar_centerpoint/include/autoware/lidar_centerpoint/postprocess/non_maximum_suppression.hpp (hunk 2 above)
Comment (technolojin) [reply perception/autoware_lidar_centerpoint/include/autoware/lidar_centerpoint/postprocess/non_maximum_suppression.hpp:38]: The logic checks the NMS type everytime, but this logic do not support multiple NMS models.
Even you have plan to implement other mode, this should come at that moment.

Comment (technolojin) [reply perception/autoware_lidar_centerpoint/include/autoware/lidar_centerpoint/postprocess/non_maximum_suppression.hpp:38]: This PR is to revise the NMS logic and improve the performance. Under this concept, removing unused parameters and logic is also included.

**Comment by kminoda** [comment]
@technolojin I'm OK with it.
(We had some PRs that changes transfusion and centerpoint at the same time, and this made the cherry-picking in some of our projects very difficult. My point here is to avoid such potential conflicts)

File: perception/autoware_lidar_centerpoint/include/autoware/lidar_centerpoint/postprocess/non_maximum_suppression.hpp (hunk 2 above)
Comment (kminoda) [reply perception/autoware_lidar_centerpoint/include/autoware/lidar_centerpoint/postprocess/non_maximum_suppression.hpp:38]: OK, understood. (Also the description update LGTM)

**Review by kminoda — APPROVED** [review]
LGTM
